### PR TITLE
Query data key changes in FP API to support multiple query datums

### PR DESCRIPTION
**Pull request by shubham1g5**
## Technical Summary

[Jira](https://dimagi-dev.atlassian.net/browse/USH-3788)

[Spec](https://docs.google.com/document/d/1hp6b3ilLFDF6YYwLLgcr18v4SNR089lOBlLrYDZMOBw/edit#heading=h.dwk4rlyousv1)

## Safety Assurance

relying on existing tests and QA automated scripts

### QA Plan

None

### Special deploy instructions

- [x] This PR can be deployed after merge with no further considerations.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations.

### Review

- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change.

cross-request: https://github.com/dimagi/commcare-core/pull/1374

### CODE REVIEW

**Review by Jtang-1 — COMMENTED** [review]
Changes make sense. But I don't understand from the changes made, when the request payload contains the query data inputs from both `query1` and `query2` screen, how does FP decide which to include in the queryResponse? Does it just include both?

**Comment by shubham1g5** [comment]
> when the request payload contains the query data inputs from both query1 and query2 screen, how does FP decide which to use for the queryResponse?

The query screen is uniquely identified by the `queryKey` now i.e. for both query1 and query2 screens, the query key must be different because of command_id and query datum id combination. If the command id is unique for 2 screens, Query datum id will need to be different for the 2 screens using the custom instance name feature.

File: src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java [hunk 1]
```diff
@@ -371,8 +370,7 @@ private Screen autoAdvanceSession(
                                 menuSession.getSessionWrapper());
                     }
                 }
-                if (previousScreen != null && previousScreen instanceof QueryScreen
-                        && FeatureFlagChecker.isToggleEnabled(TOGGLE_SPLIT_SCREEN_CASE_SEARCH)) {
+                if (previousScreen != null && previousScreen instanceof QueryScreen) {
```
[the remark this answers is not in the document]
Comment (shubham1g5) [reply src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java:373]: it's just that I needed the `queryScreen` in EntityListResponse to set the `queryKey`  in the entity list response. So I moved the ff check in EntityListResponse itself

File: src/test/java/org/commcare/formplayer/tests/CaseClaimNavigationParentChildTests.java [hunk 2]
```diff
@@ -247,12 +247,12 @@ public void testParentSkipToSearchResults(String appName, String subCaseSelectio
         testParentSearchResults(appName, queryData, selections);
         testParentSelection(appName, queryData, selections);
 
-        testChildSearchFirst(appName, queryData, new ArrayList<>(selections), "search_command.m17",
+        testChildSearchFirst(appName, new QueryData(), new ArrayList<>(selections), "search_command.m17_results",
```
[the remark this answers is not in the document]
Comment (shubham1g5) [reply src/test/java/org/commcare/formplayer/tests/CaseClaimNavigationParentChildTests.java:250]: the `queryData` was effectively empty at this point and we don't want the changes made to `queryData` in one child navigation pathway to reflect in other child navigation pathways we are passing the object to. The change just detaches the queryData objects passed to child methods from the parent copy of it.

**Review by Jtang-1 — APPROVED** [review]

**Pull request merged**